### PR TITLE
fix(v-switch): validate null as a valid value for true-value

### LINKED ISSUE #12417: [Bug Report] v-switch color not aplied when true-value is set to null
### Environment
**Vuetify Version:** 2.3.14
**Vue Version:** 2.6.12
**Browsers:** Firefox 82.0
**OS:** Windows 10

### Steps to reproduce
```javascript
<v-switch
    :false-value="true"
    :true-value="null"
    color="primary"
    v-model="switchValue"
    label="Switch button">
</v-switch>
```

### Expected Behavior
Switch should be colored with primary color

### Actual Behavior
Switch is grey.

### Reproduction Link
<a href="https://codepen.io/EkinOf/pen/bGeELVX" target="_blank">https://codepen.io/EkinOf/pen/bGeELVX</a>

### PR DESCRIPTION
**Pull request by proggreg**
## Description

fixes #12417

## Motivation and Context

Fixes an open issue, this is my first open source pull request 😬

## How Has This Been Tested?

visually

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <v-container>
    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="null"
      color="primary"
      label="Switch button"
    >
    </v-switch>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      switchValue: null,
    }),
  }
</script>

```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by johnleider** [comment]
This is a bit more involved than initially realized. There are actually quite a few scenarios are that supported with a regular `<input type="checkbox">` and a `v-model` that aren't covered by this component or even the other `selectable` components.

TLDR: vanilla `input` in Vue supports `null, undefined, 0, '', false` values as valid under certain circumstances. If we are going to update this we should address the disparity and then write unit tests for it. Here is an example playground.

<details>

```vue
<template>
  <v-container>
    <!-- <v-row hide-gutters v-for="(falseValue, i) in values" :key="i">
      <v-col v-for="(trueValue, j) in values" :key="j" cols="12">
<template v-if="falseValue !== trueValue">
          <label class="mr-6">
            true value is: <strong>{{ String(trueValue) }}</strong>
            <br>
            false value is: <strong>{{ String(falseValue) }}</strong>
          </label>
          <input type="checkbox"
          :false-value="falseValue"
          :true-value="trueValue"
          v-model="switchValue"
        >
        </template>
      </v-col>

      <v-divider class="my-4"></v-divider>
    </v-row> -->

    <v-row hide-gutters v-for="(falseValue, i) in values" :key="i">
      <v-col v-for="(trueValue, j) in values" :key="j" cols="12">
<template v-if="falseValue !== trueValue">
          <label class="mr-6">
            true value is: <strong>{{ String(trueValue) }}</strong>
            <br>
            false value is: <strong>{{ String(falseValue) }}</strong>
          </label>
          <v-checkbox
          hide-details
          class="ma-0"
          dense
          label="Checkbox"
          :false-value="falseValue"
          :true-value="trueValue"
          v-model="switchValue"
        />
        </template>
      </v-col>

      <v-divider class="my-4"></v-divider>
    </v-row>

    <!-- <input type="checkbox"
      :false-value="false"
      :true-value="''"
      v-model="switchValue"
      >
    <input type="checkbox"
      :false-value="false"
      :true-value="undefined"
      v-model="switchValue"
      >
    <input type="checkbox"
      :false-value="false"
      v-model="switchValue"
      >

    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="null"
      color="primary"
      label="Switch button"
    >
    </v-switch>
    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="''"
      color="primary"
      label="Switch button"
    >
    </v-switch> -->
    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="undefined"
      color="primary"
      label="Switch button"
    >
    </v-switch>

    {{ String(switchValue) }}
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      switchValue: null,
      values: [
      true, false, null, undefined, '', 0
    ]
    }),
  }
</script>
```

</detail>

**Review by johnleider — CHANGES_REQUESTED** [review]
This is a bit more involved than initially realized. There are actually quite a few scenarios are that supported with a regular `<input type="checkbox">` and a `v-model` that aren't covered by this component or even the other `selectable` components.

TLDR: vanilla `input` in Vue supports `null, undefined, 0, '', false` values as valid under certain circumstances. If we are going to update this we should address the disparity and then write unit tests for it. Here is an example playground.

<details>

```vue
<template>
  <v-container>
    <!-- <v-row hide-gutters v-for="(falseValue, i) in values" :key="i">
      <v-col v-for="(trueValue, j) in values" :key="j" cols="12">
<template v-if="falseValue !== trueValue">
          <label class="mr-6">
            true value is: <strong>{{ String(trueValue) }}</strong>
            <br>
            false value is: <strong>{{ String(falseValue) }}</strong>
          </label>
          <input type="checkbox"
          :false-value="falseValue"
          :true-value="trueValue"
          v-model="switchValue"
        >
        </template>
      </v-col>

      <v-divider class="my-4"></v-divider>
    </v-row> -->

    <v-row hide-gutters v-for="(falseValue, i) in values" :key="i">
      <v-col v-for="(trueValue, j) in values" :key="j" cols="12">
<template v-if="falseValue !== trueValue">
          <label class="mr-6">
            true value is: <strong>{{ String(trueValue) }}</strong>
            <br>
            false value is: <strong>{{ String(falseValue) }}</strong>
          </label>
          <v-checkbox
          hide-details
          class="ma-0"
          dense
          label="Checkbox"
          :false-value="falseValue"
          :true-value="trueValue"
          v-model="switchValue"
        />
        </template>
      </v-col>

      <v-divider class="my-4"></v-divider>
    </v-row>

    <!-- <input type="checkbox"
      :false-value="false"
      :true-value="''"
      v-model="switchValue"
      >
    <input type="checkbox"
      :false-value="false"
      :true-value="undefined"
      v-model="switchValue"
      >
    <input type="checkbox"
      :false-value="false"
      v-model="switchValue"
      >

    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="null"
      color="primary"
      label="Switch button"
    >
    </v-switch>
    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="''"
      color="primary"
      label="Switch button"
    >
    </v-switch> -->
    <v-switch
      v-model="switchValue"
      :false-value="false"
      :true-value="undefined"
      color="primary"
      label="Switch button"
    >
    </v-switch>

    {{ String(switchValue) }}
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      switchValue: null,
      values: [
      true, false, null, undefined, '', 0
    ]
    }),
  }
</script>
```

</detail>

**Comment by MajesticPotatoe** [comment]
pulse check

**Comment by MajesticPotatoe** [comment]
This PR is being closed due to inactivity.

If you have any additional questions, please reach out to us in our [Discord community](https://community.vuetifyjs.com/).
